content/Solutions/Optimizations/Networking.md

```diff
@@ -0,0 +1,339 @@
+
+### Network Testing
+
+Use [iperf3](https://software.es.net/iperf/index.html) to test the max bandwith between the TrueNAS system and a client computer.
+iperf comes installed in TrueNAS CORE and SCALE.
+Before you begin, check the client computer and install iperf if needed.
+
+Enter {{< cli >}}iperf3 -s{{< /cli >}} on the TrueNAS system.
+This tells the TrueNAS system that it is acting as server/host and activates the iperf listener.
+The active port displays:
+```
+----------------------------------
+Server listening on 5201 (test #1)
+----------------------------------
+```
+If you want to specify a port to use, you can activate iperf with {{< cli >}}iperf3 -s -p *5101*{{< /cli >}}, where *5101* is the port to monitor. See [Default Ports](https://www.truenas.com/docs/references/defaultports/) for a list of assigned port numbers.
+
+Next, open a CLI on the client computer and enter {{< cli >}}iperf3 -c *hostname* -p *5201*{{< /cli >}}, where *hostname* is the IP address or hostname and domain for the host server and *5201* is the port the server is listening on.
+By default, the iperf test runs for 10 seconds and outputs transfer rates, for example:
+
+```
+Connecting to host 8.8.8.8, port 5201
+[ 5] local 8.8.8.8 port 44706 connected to 8.8.8.8 port 5201
+[ ID] Interval           Transfer     Bitrate
+[  5]   0.00-1.00   sec  52.9 MBytes   444 Mbits/sec                  
+[  5]   1.00-2.00   sec  55.5 MBytes   465 Mbits/sec                  
+[  5]   2.00-3.00   sec  55.5 MBytes   465 Mbits/sec                  
+[  5]   3.00-4.00   sec  53.4 MBytes   448 Mbits/sec                  
+[  5]   4.00-5.00   sec  53.6 MBytes   450 Mbits/sec                  
+[  5]   5.00-6.00   sec  58.5 MBytes   491 Mbits/sec                  
+[  5]   6.00-7.00   sec  57.3 MBytes   481 Mbits/sec                  
+[  5]   7.00-8.00   sec  58.5 MBytes   491 Mbits/sec                  
+[  5]   8.00-9.00   sec  56.6 MBytes   475 Mbits/sec                  
+[  5]   9.00-10.00  sec  59.1 MBytes   495 Mbits/sec                  
+[  5]  10.00-10.00  sec   191 KBytes   538 Mbits/sec                  
+- - - - - - - - - - - - - - - - - - - - - - - - -
+[ ID] Interval           Transfer     Bitrate            Retr
+- - - - - - - - - - - - - - - - - - - - - - - - -
+[  5]   0.00-10.00  sec   562 MBytes   471 Mbits/sec     0            sender
+[  5]   0.00-10.00  sec   561 MBytes   471 Mbits/sec                  receiver
+
+iperf Done.
+```
+
+iperf 3 is single threaded, which means that some hosts may be CPU-bound, including 40G and 100G networks.
+See the [iperf FAQ](https://software.es.net/iperf/faq.html) for more information
+
+To run parallel streams of iperf3 on multiple cores/ports, first initialize the TrueNAS system on multiple ports:
+
+{{< cli >}}iperf3 -s -p *5101*&; iperf3 -s -p *5102*&; iperf3 -s -p *5103* &;{{< /cli >}}
+
+Next, run multiple instances on the client system, using the "-T" flag to label the output:
+
```

Comment:
Might be worth mentioning -P flag to run parallel streams. Far easier and will often be the recomended option on the forums, reddit, etc.

Reply:
As far as I understand it, the -P flag is still single threaded in iperf3 (at least the version in TrueNAS, they added multithreadding for parallel tests in v 3.16) so might not give an accurate picture. Could mention it with that caveat though?

Reply:
I think it is worth mentioning, for home users it should be adequate.